openfl/experimental/workflow/component/director/director.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+"""Experimental Director module."""
+
+import asyncio
+import logging
+import time
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, AsyncGenerator, Dict, Iterable, Optional, Tuple, Union
+
+import dill
+
+from openfl.experimental.workflow.component.director.experiment import (
+    Experiment,
+    ExperimentsRegistry,
+)
+from openfl.experimental.workflow.transport.grpc.exceptions import EnvoyNotFoundError
+
+logger = logging.getLogger(__name__)
+
+
+class Director:
+    """Director class for managing experiments and envoys.
+
+    Attributes:
```

Comment:
I have to admit I haven't very often seen the attributes of a class listed in a docstring... I'd suggest reconsidering this approach as such a docstring is difficult to maintain (and redundant with the code IMHO).